### ISSUE TITLE
MigrationCorrectnessTest.testPartitionData_whenBackupNodesStartedTerminated[backups:3,nodes:7]

### ISSUE THREAD

**Issue by hasancelik** [hazelcast/hazelcast#8076]
```
java.lang.AssertionError: null
    at org.junit.Assert.fail(Assert.java:86)
    at org.junit.Assert.assertTrue(Assert.java:41)
    at org.junit.Assert.assertTrue(Assert.java:52)
    at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.startNodes(PartitionCorrectnessTestSupport.java:96)
    at com.hazelcast.internal.partition.MigrationCorrectnessTest.testPartitionData_whenBackupNodesStartedTerminated(MigrationCorrectnessTest.java:113)
    at com.hazelcast.internal.partition.MigrationCorrectnessTest.testPartitionData_whenBackupNodesStartedTerminated(MigrationCorrectnessTest.java:95)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x/com.hazelcast$hazelcast/4978/testReport/junit/com.hazelcast.internal.partition/MigrationCorrectnessTest/testPartitionData_whenBackupNodesStartedTerminated_backups_3_nodes_7_/

**Comment by mdogan** [comment]
most probably a mock joiner issue. see https://github.com/hazelcast/hazelcast/issues/7463

**Comment by metanet** [comment]
Most probably fixed by recent changes in MockJoiner. Closing this one.

**Comment by hasancelik** [comment]
**testPartitionData_whenBackupNodesStartedTerminated**
```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.startNodes(PartitionCorrectnessTestSupport.java:98)
	at com.hazelcast.internal.partition.AbstractMigrationCorrectnessTest.testPartitionData_whenBackupNodesStartedTerminated(AbstractMigrationCorrectnessTest.java:90)
	at com.hazelcast.internal.partition.AbstractMigrationCorrectnessTest.testPartitionData_whenBackupNodesStartedTerminated(AbstractMigrationCorrectnessTest.java:72)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)
	at java.lang.reflect.Method.invoke(Method.java:597)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:88)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:82)
	at java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)
	at java.util.concurrent.FutureTask.run(FutureTask.java:138)
	at java.lang.Thread.run(Thread.java:662)
```

https://hazelcast-l337.ci.cloudbees.com/view/Official%20Builds/job/Hazelcast-3.x-OracleJDK1.6/com.hazelcast$hazelcast/1015/testReport/junit/com.hazelcast.internal.partition/MigrationCorrectnessTest/testPartitionData_whenBackupNodesStartedTerminated_backups_1_nodes_2_/

**Comment by hasancelik** [comment]
```
java.lang.AssertionError: null
	at org.junit.Assert.fail(Assert.java:86)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at org.junit.Assert.assertTrue(Assert.java:52)
	at com.hazelcast.internal.partition.PartitionCorrectnessTestSupport.terminateNodes(PartitionCorrectnessTestSupport.java:146)
	at com.hazelcast.internal.partition.AbstractMigrationCorrectnessTest.testPartitionData_whenBackupNodesTerminated(AbstractMigrationCorrectnessTest.java:66)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:88)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:82)
	at java.util.concurrent.FutureTask.run(FutureTask.java:262)
	at java.lang.Thread.run(Thread.java:745)
```

https://hazelcast-l337.ci.cloudbees.com/view/Hazelcast/job/Hazelcast-3.x-OracleJDK1.7/com.hazelcast$hazelcast/680/testReport/junit/com.hazelcast.internal.partition/MigrationCorrectnessTest/testPartitionData_whenBackupNodesTerminated_backups_3_nodes_4_/

**Comment by mdogan** [comment]
already fixed by #9323
